### PR TITLE
Add summary formatter section

### LINKED ISSUE #2654: Add information on the summaryFormatter property
Adding description about the usage of the summaryFormatter property [#8786](https://github.com/IgniteUI/igniteui-angular/issues/8786)

### PR DESCRIPTION
**Pull request by mtsvyatkova**
Closes #2654 

### Checklist:
 
 - [ ] check topic's TOC/menu and paragraph headings
 - [ ] link to other topics using `../relative/path.md`
 - [ ] at the References section at the end of the topic add links to topics, samples, etc
 - [ ] reference API documentation instead of adding a section with API

<br />

 - [ ] use valid component names - [Data] Grid, `IgxSelectComponent`, `<igx-combo>`
 - [ ] use spell checker tool (VS Code, Grammarly, Microsoft Editor)
 - [ ] add inline `code blocks` for the names of classes / tags / properties
 - [ ] add language descriptor for the ```code blocks```
 - [ ] check broken links (use browser add-on)
 - [ ] check if sample is working and fully visible in the topic
 - [ ] check if sample is working and fully visible in the StackBlitz
 - [ ] check if code blocks match the code in StackBlitz demo

<br />

 - [ ] follow SEO guidelines to fill topic's metadata ([SEO guidelines](https://infragisticsinc297.sharepoint.com/:w:/g/Groups/marketing/EWz9InT4FDlErHCumxsKGY4Bd8H03yhRWxDFk47luRz-_Q?e=S5wWcx))

<br />

 - [ ] do not resolve requested changes (leave that to the reviewer)
 - [ ] add `pending-localization` label when the review of the PR is done
 - [ ] add a member from the localization team to translate it

### CODE REVIEW

**Comment by gedinakova** [comment]
@mtsvyatkova The sample in Hierarchical grid topic is not loaded because of the following errors:
igniteui-theming-widget.js?t=?t=1612430854782:2 GET http://localhost:4200/assets/samples/hierarchical-grid--hierarchical-grid-summary-formatter.json 404 (Not Found)
_ @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleMacroTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
l @ igniteui-theming-widget.js?t=?t=1612430854782:2
(anonymous) @ igniteui-theming-widget.js?t=?t=1612430854782:2
D.s.<computed> @ igniteui-theming-widget.js?t=?t=1612430854782:2
send @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
ajax @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
h.each.h.<computed> @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
(anonymous) @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
(anonymous) @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
(anonymous) @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
c @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
fireWith @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
x @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
(anonymous) @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
T @ igniteui-theming-widget.js?t=?t=1612430854782:2
invokeTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
runTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
invokeTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
p @ igniteui-theming-widget.js?t=?t=1612430854782:2
f @ igniteui-theming-widget.js?t=?t=1612430854782:2
load (async)
P @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleEventTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
(anonymous) @ igniteui-theming-widget.js?t=?t=1612430854782:2
E.s.set @ igniteui-theming-widget.js?t=?t=1612430854782:2
send @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
ajax @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
h.each.h.<computed> @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
h @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
(anonymous) @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
c @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
fireWith @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
x @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
(anonymous) @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
T @ igniteui-theming-widget.js?t=?t=1612430854782:2
invokeTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
runTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
invokeTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
p @ igniteui-theming-widget.js?t=?t=1612430854782:2
f @ igniteui-theming-widget.js?t=?t=1612430854782:2
load (async)
P @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
scheduleEventTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
(anonymous) @ igniteui-theming-widget.js?t=?t=1612430854782:2
E.s.set @ igniteui-theming-widget.js?t=?t=1612430854782:2
send @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
ajax @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
h.each.h.<computed> @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
p @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
(anonymous) @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
(anonymous) @ scripts-bundle.min.js?checksum=392646447158c1e332650e4d73c280ac:7
c @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
fireWith @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
ready @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
M @ docfx-bundle.min.js?checksum=e0ba925bf63c64e33693368efb48380d:7
invokeTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
runTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
invokeTask @ igniteui-theming-widget.js?t=?t=1612430854782:2
p @ igniteui-theming-widget.js?t=?t=1612430854782:2
f @ igniteui-theming-widget.js?t=?t=1612430854782:2
vendor.js:179835 ERROR Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'hierarchical-grid/hierarchical-grid-summary-formatter'
Error: Cannot match any routes. URL Segment: 'hierarchical-grid/hierarchical-grid-summary-formatter'
    at ApplyRedirects.noMatchError (vendor.js:253539)
    at CatchSubscriber.selector (vendor.js:253513)
    at CatchSubscriber.error (vendor.js:18321)
    at MapSubscriber._error (vendor.js:13929)
    at MapSubscriber.error (vendor.js:13898)
    at MapSubscriber._error (vendor.js:13929)
    at MapSubscriber.error (vendor.js:13898)
    at ThrowIfEmptySubscriber._error (vendor.js:13929)
    at ThrowIfEmptySubscriber.error (vendor.js:13898)
    at TakeLastSubscriber._error (vendor.js:13929)
    at resolvePromise (polyfills.js:1720)
    at resolvePromise (polyfills.js:1663)
    at polyfills.js:1789
    at ZoneDelegate.invokeTask (polyfills.js:1170)
    at Object.onInvokeTask (vendor.js:207201)
    at ZoneDelegate.invokeTask (polyfills.js:1170)
    at Zone.runTask (polyfills.js:920)
    at drainMicroTaskQueue (polyfills.js:1386)

File: en/components/grids_templates/summaries.md [hunk 1]
```diff
@@ -228,7 +228,7 @@ interface IgxSummaryResult {
 }
 ```
 and take optional parameters for calculating the summaries.
-See [Custom summaries, which access all data](#custom-summaries-which-access-all-grid-data) and [Custom summaries with localization](#custom-summaries-with-localization) sections below.
+See [Custom summaries, which access all data](#custom-summaries-which-access-all-grid-data) section below.
```
Comment (gedinakova) [inline en/components/grids_templates/summaries.md:231]: This link doesn't work in Tree Grid and Hierarchical Grid topics (it is specific to the Grid topic).

File: en/components/grids_templates/summaries.md [hunk 2]
```diff
@@ -285,7 +285,7 @@ export class HGridSummarySampleComponent implements OnInit {
 ```
 }
 
-### Custom summaries, which access all @@igComponent data
+## Custom summaries, which access all @@igComponent data
  Now you can access all grid data inside the custom column summary. Two additional optional parameters are introduced in the IgxSummaryOperand `operate` method.
```
Comment (gedinakova) [inline en/components/grids_templates/summaries.md:289]: Maybe "grid" should be "@@igComponent".

**Review by gedinakova — CHANGES_REQUESTED** [review]
The link

File: en/components/grids_templates/summaries.md [hunk 3]
```diff
@@ -307,47 +307,75 @@ class MySummary extends IgxNumberSummaryOperand {
...
+           data-demos-base-url="{environment:demosBaseUrl}"
            iframe-src="{environment:demosBaseUrl}/hierarchical-grid/hierarchical-grid-allData-summary" >
 </code-view>
 
 }
 
-### Custom summaries with localization
-By default, summary results are localized and formatted according to the grid [`locale`]({environment:angularApiUrl}/classes/igxgridcomponent.html#locale) and column [`pipeArgs`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#pipeArgs). When you are providing custom summaries, you may want to format the result in the corresponding localization and format, thus to match the values in the column.
+## Format summaries
+By default, summary results are localized and formatted according to the grid [`locale`]({environment:angularApiUrl}/classes/igxgridcomponent.html#locale) and column [`pipeArgs`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#pipeArgs). When you want to change the default appearance of the summaries, you may format the result with the [`summaryFormatter`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#summaryFormatter) property.
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/summaries.md:334]: We should specify that only the built-in summary operands are formatted using the `locale` and `pipeArgs`. If you have custom summary operands even if they derive from our classes we won't format the summary results.

File: en/components/grids_templates/summaries.md [hunk 4]
```diff
@@ -307,47 +307,75 @@ class MySummary extends IgxNumberSummaryOperand {
...
+<code-view style="height:650px"
+           data-demos-base-url="{environment:demosBaseUrl}"
            iframe-src="{environment:demosBaseUrl}/hierarchical-grid/hierarchical-grid-allData-summary" >
 </code-view>
 
 }
 
-### Custom summaries with localization
-By default, summary results are localized and formatted according to the grid [`locale`]({environment:angularApiUrl}/classes/igxgridcomponent.html#locale) and column [`pipeArgs`]({environment:angularApiUrl}/classes/igxcolumncomponent.html#pipeArgs). When you are providing custom summaries, you may want to format the result in the corresponding localization and format, thus to match the values in the column.
+## Format summaries
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/summaries.md:333]: Rename to `Summaries Formatter` maybe?

**Review by DiyanDimitrov — CHANGES_REQUESTED** [review]
We should mark the Summaries topic as updated.

File: en/components/grids_templates/summaries.md [hunk 5]
```diff
@@ -307,47 +307,75 @@ class MySummary extends IgxNumberSummaryOperand {
...
-class MySummary extends IgxDateSummaryOperand {
-    operate(columnData: any[], allData = [], fieldName, locale: string, pipeArgs: IColumnPipeArgs): IgxSummaryResult[] {
-        const pipeArgs: IColumnPipeArgs = {
-            format: 'longDate',
-            timezone: 'UTC',
-            digitsInfo: '1.1-2'
-        }
-        const result = super.operate(columnData, allData, fieldName, locale, pipeArgs);
-        return result;
+public dateSummaryFormat(summary: IgxSummaryResult, summaryOperand: IgxSummaryOperand): string {
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/summaries.md:337]: We could probably add a code snippet for the html for example:
```html
<igx-column ... [summaryFormatter]="dateSummaryFormat"></igx-column>
```

**Review by DiyanDimitrov — APPROVED** [review]

**Review by gedinakova — APPROVED** [review]

**Pull request merged**